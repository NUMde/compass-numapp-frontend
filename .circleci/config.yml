# Use the latest 2.1 version of CircleCI pipeline process engine.
# See: https://circleci.com/docs/configuration-reference
# For a detailed guide to building and testing on iOS, read the docs:
# https://circleci.com/docs/testing-ios/
version: 2.1

jobs:
  node:
    working_directory: ~/compass-template
    docker: 
        - image: cimg/node:18.12
    steps: 
      - checkout

      - run: cd ./compass-numapp-template
      - restore_cache:
          key: npm-v1-{{ checksum "./compass-numapp-template/package-lock.json" }}-{{ arch }}

      - restore_cache:
          key: node-v1-{{ checksum "./compass-numapp-template/package.json" }}-{{ arch }}

      - run: cd ~ && npx react-native init compass --template ~/compass-template --title "COMPASS"
      
      - save_cache:
          key: npm-v1-{{ checksum "./compass-numapp-template/package-lock.json" }}-{{ arch }}
          paths:
            - ~/.cache/npm

      - save_cache:
          key: node-v1-{{ checksum "./compass-numapp-template/package.json" }}-{{ arch }}
          paths:
            - ./compass/node_modules
      - persist_to_workspace:
          root: ../compass
          paths:
            - '*'
  build_ios:
    working_directory: ~/compass/ios
    # execution environment.
    # See: https://circleci.com/docs/configuration-reference/#docker-machine-macos-windows-executor
    macos:
      xcode: 14.2.0
    environment:
      CI: 'true'

    # See: https://circleci.com/docs/configuration-reference/#steps
    steps:
      - attach_workspace: 
          at: ~/compass/
      - run: pod install
      - run: fastlane scan # Run tests using Fastlane

      # Collect XML test results data to show in the UI, and save the same XML
      # files under test-results folder in the Artifacts tab
      - store_test_results:
          path: test_output
      - store_artifacts:
          path: test_output
          destination: scan-output

# Invoke jobs via workflows
# See: https://circleci.com/docs/configuration-reference/#workflows
workflows:
  build: # This is the name of the workflow, feel free to change it to better match your workflow.
    # Inside the workflow, you define the jobs you want to run.
    jobs:
      - node
      - build_ios:
          requires: 
            - node

    # - build_android
    #     requires:
    #       - node
