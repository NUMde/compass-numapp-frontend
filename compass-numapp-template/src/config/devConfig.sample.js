import { name as appName } from '../../appName.json';

export default {
  // DEVELOPMENT OPTIONS
  /*-----------------------------------------------------------------------------------*/

  // these options are meant for development only.

  /** dev-option:
   * skips the QR login after 500ms */
  automateQrLogin: __DEV__ && false,

  /** dev-option:
   * subject-id for automated login */
  automateQrLoginSubjectId: `{"AppIdentifier":"${appName}","SubjectId":"7bfc3b07-a97d-4e11-8ac6-b970c1745476"}`,

  /** dev-option:
   * shows a button to erase all data (in the about-menu) - dev-only */
  showEraseAll: __DEV__ && true,

  /** dev-option:
   * shows a button to log out the user (in the about-menu) - dev-only */
  showLogout: __DEV__ && true,

  /** dev-option:
   * logs out the response-json parsed as an object in the developer console
   * */
  logPureResponse: __DEV__ && false,

  /** dev-option:
   * logs out the response-json in the developer console
   * */
  logPureResponseJSON: __DEV__ && false,

  /** dev-option:
   * logs out the encrypted response-json in the developer console
   * */
  logEncryptedResponse: __DEV__ && false,

  /** dev-option:
   * used the locally available questionnaire.js instead of the procured one from the backend */
  useLocalQuestionnaireInsteadOfTheReceivedOne: __DEV__ && false,

  /** dev-option:
   * skips the id-comparison */
  skipIncomingQuestionnaireCheck: __DEV__ && false,

  // development-base-backend-uri
  baseUri: 'http://127.0.0.1:8080/api/',
};
