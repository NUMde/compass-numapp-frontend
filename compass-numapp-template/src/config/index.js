import theme from './theme';
import text from './textConfig';
import appConfig from './appConfig';

/** combine the default values with the custom values */
export {
  /** provides variables controlling the style of the application */
  theme,
  /** provides the text-strings visible the application */
  text as strings,
  /** provides options to alter the behaviour of the app */
  appConfig,
};
