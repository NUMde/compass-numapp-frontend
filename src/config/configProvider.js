import defaultThemeValues from '../theme/theme';
import defaultTextValues from './textConfig';
import defaultAppConfigValues from './appConfig';

import defaultThemeValues from '../theme/theme';
import defaultAppConfigValues from './appConfig';

import customThemeValues from '../CUSTOMIZATION/customTheme';
import customAppConfigValues from '../CUSTOMIZATION/customAppConfig';

let exportValues  = {

    /** provides variables controlling the style of the application */
    theme: {
        ...defaultThemeValues,
        ...customThemeValues
    },

    /** provides options to alter the behaviour of the app */
    appConfig: {
        ...defaultAppConfigValues,
        ...customAppConfigValues
    },
};

/** combine the default values with the custom values */
export default exportValues
