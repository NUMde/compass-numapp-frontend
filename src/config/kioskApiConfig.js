/***********************************************************************************************
imports
***********************************************************************************************/

// the predefined questionnaire the demo is gonna use
import hardcodedTestQuestionnaire from "../assets/files/questionnaire";

/***********************************************************************************************
 constants
 ***********************************************************************************************/

 // is used to keep track of the completed transmissions (faked ones)
let playthrough = 1;

// is used to determine if a report was just sent out
let sentOutReport = false;

// is used to keep track if the first response was already sent out
let sentOutTheFirstOne = false;

// default user data
const defaultMockUserData = {
    firstTime: false,
    current_interval: 7,
    additional_iterations_left: 0,
    current_questionnaire_id: "final",
    due_date: "9999-12-30T23:00:00.000Z",
    start_date: "2021-07-15T04:00:00.000Z",
    subjectId: "7bfc3b07-a97d-4e11-8ac6-b970c1745476",
    pushAppGUID: "f98fc829-1d4d-418b-b05b-a147fdb9c0f3",
    current_instance_id: "a177e102-ecfa-490e-81a7-04cf7ebf22cf",
    pushClientSecret: "ab4b7d62-1fde-4a46-a566-2d1ada678df3",
    recipient_certificate_pem_string:
        "-----BEGIN CERTIFICATE-----\nMIIEOzCCAyOgAwIBAgIUc0iY16RQuluxpRL7hsmq/FRaWt0wDQYJKoZIhvcNAQEL\nBQAwgawxCzAJBgNVBAYTAkRFMQwwCgYDVQQIDANSTFAxDjAMBgNVBAcMBU1haW56\nMREwDwYDVQQKDAhVTSBNYWluejEeMBwGA1UECwwVUHJldmVudGl2ZSBDYXJkaW9s\nb2d5MRUwEwYDVQQDDAxHQ1NTdHVkeUNlcnQxNTAzBgkqhkiG9w0BCQEWJnN2ZW4t\nb2xpdmVyLnRyb2Vic0B1bmltZWRpemluLW1haW56LmRlMB4XDTIwMDkwMTExMjk0\nNFoXDTIzMDUyOTExMjk0NFowgawxCzAJBgNVBAYTAkRFMQwwCgYDVQQIDANSTFAx\nDjAMBgNVBAcMBU1haW56MREwDwYDVQQKDAhVTSBNYWluejEeMBwGA1UECwwVUHJl\ndmVudGl2ZSBDYXJkaW9sb2d5MRUwEwYDVQQDDAxHQ1NTdHVkeUNlcnQxNTAzBgkq\nhkiG9w0BCQEWJnN2ZW4tb2xpdmVyLnRyb2Vic0B1bmltZWRpemluLW1haW56LmRl\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnmY9Ep2SIoo824dWk1y8\ngonKArnhwVyW3AAVLpz9UtmkfElLiNLUbVOv6dGyCG9W8Y0DFwUU33EmFBlsmX3R\n7LbnwLYn7R6ntSakr58P6yNjMXY2CT2GjI0m63DPbLh6S3J6cO/8HNRnL3zW4O9B\n6FtgVg7Yf2koFjXz/kUYOFeqsPoqqLk70pO1nPGTjkz2JhBJQ+SaulTo19ZQ2tkT\nVP3zTZ9Urk8i/3fCU1TKTbYL/WwQepJlN8BwxzjIDQ2wA62nXzzlHQN98MnQGgY4\nXvj679rLFLsz2vgVmLnGPFwFL5op/6gmsKz7Q+B3zGco6qGyyL9N/D4fLKOVxnMy\nIwIDAQABo1MwUTAdBgNVHQ4EFgQUyMC1pkSWjQbIFZ/o4SEV9IqHJmwwHwYDVR0j\nBBgwFoAUyMC1pkSWjQbIFZ/o4SEV9IqHJmwwDwYDVR0TAQH/BAUwAwEB/zANBgkq\nhkiG9w0BAQsFAAOCAQEAIAAhhtVpDf/767pg8xxwaETYjPRL4pKW4H5b9t2Vr41M\nAdCUou9dbtWlqeFvY/eBby9DM05lI+D8hX2wYVRr7Pc80fuw5k4f+uTry2jc0vFV\nvHhv9sHJq3+EX/P4Nckb7M5lKI+d6GDiqzU3r/ARstoRVbQHRu2f0W7SW0pVPwlQ\nSFv61NIkjQB19SAO1IAR4Z/UrZ7Y0vnKQZ2fqFUHoD5wz4jXhHFsungocg9IxJ74\nKXaMj1nTtjCGxx11jQCNLyrVjI/XgNoPQ1v2FIQxmPJju9MAXQ2AIPA2NTXaDXHk\nYDlCrh7jxSgVWfrijhMFO/BmHDlumBpOPO1S4XG4qA==\n-----END CERTIFICATE-----",
};

/***********************************************************************************************
methods
***********************************************************************************************/

/**
 * calculates the next start date
 * @param  {boolean} laterOn tells us if we need to up the date
 */
const getStartDate = (laterOn) => {
    const now = new Date;
    now.setDate(now.getDate() + (laterOn ? 3 * playthrough : 0));
    return now.toISOString()
};

/**
 * calculates the next due date
 * @param  {boolean} laterOn tells us if we need to up the date
 */
const getDueDate = (laterOn) => {
    const then = new Date
    then.setDate(then.getDate() + (laterOn ? 10 * playthrough : 7));
    return then.toISOString()
};

/**
 * generates a fitting set of userdata
 */
const generateMockUserData = () => {
    // first time user
    if(!sentOutTheFirstOne) return {
        ...defaultMockUserData,
        current_interval: playthrough,
        start_date: getStartDate(),
        due_date: getDueDate(),
        firstTime: true,
    };
    // after sending out a regular questionnaire
    if(!sentOutReport) return {
        ...defaultMockUserData,
        current_interval: playthrough,
        start_date: getStartDate(true),
        due_date: getDueDate(true),
        firstTime: false,
    };
    // after sending out a report
    return {
        ...defaultMockUserData,
        current_interval: playthrough,
        start_date: getStartDate(),
        due_date: getDueDate(),
        firstTime: false,
    };
};

/**
 * triggers the mock for the sendQuestionnaire call and updates the status variables
 */
const sendQuestionnaire = () => {
    sentOutTheFirstOne = true;
    sentOutReport = false;
    playthrough += 1;
    return Promise.resolve(generateMockUserData());
};

/**
 * triggers the mock for the sendReport call and updates a status variable
 */
const sendReport = () => {
    sentOutReport = true;
    return Promise.resolve(generateMockUserData());
};

/***********************************************************************************************
export
***********************************************************************************************/

export default {
    sendReport,
    sendQuestionnaire,
    updateDeviceToken: () => Promise.resolve({ data: "something something" }),
    login: () => Promise.resolve({ data: generateMockUserData() }),
    getUserUpdate: () => Promise.resolve({ data: generateMockUserData() }),
    getBaseQuestionnaire: () => Promise.resolve({ data: hardcodedTestQuestionnaire })
};
