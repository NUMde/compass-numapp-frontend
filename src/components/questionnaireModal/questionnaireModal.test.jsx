import React from 'react';

import {
  renderWithRedux,
  act,
  fireEvent,
  within,
  waitFor,
} from '../../../__test-utils__/render';
import emptyItemMap from '../../../__mocks__/questionnaire/emptyItemMap';
import categories from '../../../__mocks__/questionnaire/categories';

import config from '../../config/configProvider';

import QuestionnaireModal from './questionnaireModal';

const initialState = {
  Questionnaire: {
    itemMap: emptyItemMap,
    categories,
    categoryIndex: 0,
    pageIndex: 1,
  },
};

describe('Questionnaire Modal', () => {
  it("should render the questionnaireModal in the 'hidden state' (no item,no BottomBar)", () => {
    const { queryByTestId, getByTestId } = renderWithRedux(
      <QuestionnaireModal />,
    );
    expect(getByTestId('QuestionnaireModal').props.visible).toBe(false);
    expect(queryByTestId(/QuestionnaireItem/)).toBeFalsy();
    expect(queryByTestId('BottomBar')).toBeFalsy();
  });

  it('should render the questionnaireModal with first item of first category', () => {
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState,
    });
    expect(getByTestId('QuestionnaireModal').props.visible).toBe(true);
    expect(getByText(/Freitextabfrage/)).toBeTruthy();
    expect(getByTestId('BottomBar')).toBeTruthy();
  });

  it("when answer is provided, 'confirm button' should change appearance", async () => {
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState,
    });
    expect(getByTestId('QuestionnaireModal').props.visible).toBe(true);
    expect(getByText(/Freitextabfrage/)).toBeTruthy();
    const input = getByTestId('BasicInput.Input');
    expect(input).toBeTruthy();
    const icon = within(getByTestId('BottomBar_confirm_btn')).getByTestId(
      'iconIcon',
    );
    expect(icon.parent.parent.parent.props.style.backgroundColor).toBe(
      config.theme.colors.accent4,
    );
    act(() => {
      fireEvent.changeText(input, 'some text');
    });
    expect(input.props.value).toBe('some text');
    await waitFor(() =>
      expect(icon.parent.parent.parent.props.style.backgroundColor).toBe(
        config.theme.colors.success,
      ),
    );
  });

  it("should navigate to next question when clicking 'confirm button'", async () => {
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState,
    });
    const confirmButton = getByTestId('BottomBar_confirm_btn');
    act(() => fireEvent.press(confirmButton));

    await waitFor(() => expect(getByText(/Datumsabfrage/)).toBeTruthy());
    expect(getByTestId('chosenDate')).toBeTruthy();
  });

  it("should navigate to next question when clicking 'forward button'", async () => {
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState,
    });
    const confirmButton = getByTestId('BottomBar_fwd_btn');
    act(() => fireEvent.press(confirmButton));

    await waitFor(() => expect(getByText(/Datumsabfrage/)).toBeTruthy());
    expect(getByTestId('chosenDate')).toBeTruthy();
  });

  it("should navigate to previous question when clicking 'back button'", async () => {
    const modifiedInitialState = { ...initialState };
    modifiedInitialState.Questionnaire.pageIndex = 2;
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState,
    });
    () => expect(getByText(/Datumsabfrage/)).toBeTruthy();
    const backButton = getByTestId('BottomBar_back_btn');
    act(() => fireEvent.press(backButton));

    await waitFor(() => expect(getByText(/Freitext/)).toBeTruthy());
  });

  it("should close the modal when clicking 'close button'", async () => {
    const { getByTestId, queryByTestId } = renderWithRedux(
      <QuestionnaireModal />,
      {
        initialState,
      },
    );
    const closeButton = getByTestId('QuestionnaireModal_close');
    act(() => fireEvent.press(closeButton));

    await waitFor(() =>
      expect(getByTestId('QuestionnaireModal').props.visible).toBe(false),
    );
    expect(queryByTestId('BottomBar')).toBeFalsy();
  });

  it("should close the modal when clicking 'confirm button' on last page", async () => {
    const modifiedInitialState = { ...initialState };
    modifiedInitialState.Questionnaire.pageIndex = 23;
    const { getByTestId, queryByTestId } = renderWithRedux(
      <QuestionnaireModal />,
      {
        initialState: modifiedInitialState,
      },
    );

    const confirmButton = getByTestId('BottomBar_confirm_btn');
    act(() => fireEvent.press(confirmButton));

    await waitFor(() =>
      expect(getByTestId('QuestionnaireModal').props.visible).toBe(false),
    );
    expect(queryByTestId('BottomBar')).toBeFalsy();
  });

  it('should skip questions whose dependencies are not met', async () => {
    const modifiedInitialState = { ...initialState };
    modifiedInitialState.Questionnaire.pageIndex = 11;
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState: modifiedInitialState,
    });
    expect(getByText(/Choice-Abfrage/)).toBeTruthy();
    const confirmButton = getByTestId('BottomBar_confirm_btn');
    act(() => fireEvent.press(confirmButton));

    await waitFor(() => expect(getByText(/Abfrage Dezimalzahl/)).toBeTruthy());
  });

  it('should skip questions whose dependencies are not met (backwards)', async () => {
    const modifiedInitialState = { ...initialState };
    modifiedInitialState.Questionnaire.pageIndex = 15;
    const { getByTestId, getByText } = renderWithRedux(<QuestionnaireModal />, {
      initialState: modifiedInitialState,
    });
    expect(getByText(/Abfrage Dezimalzahl/)).toBeTruthy();
    const backButton = getByTestId('BottomBar_back_btn');
    act(() => fireEvent.press(backButton));

    await waitFor(() => expect(getByText(/Choice-Abfrage/)).toBeTruthy());
  });
});
