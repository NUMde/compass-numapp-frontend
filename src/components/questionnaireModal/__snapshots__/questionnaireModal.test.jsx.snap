// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questionnaire Modal should render the modal with the progressBar 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  onSwipeComplete={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={50}
  scrollTo={[Function]}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  testID="QuestionnaireModal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "borderRadius": 10,
            "justifyContent": "flex-end",
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 20,
          },
          Object {
            "margin": 0,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={50}
      scrollTo={[Function]}
      statusBarTranslucent={false}
      style={
        Object {
          "borderRadius": 10,
          "flex": 1,
          "justifyContent": "flex-end",
          "left": 0,
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 20,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeDirection={
        Array [
          "down",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "backgroundColor": "#F8F9FA",
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": "auto",
            "maxHeight": "90%",
            "padding": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 10,
              "width": "auto",
            }
          }
        >
          <Text
            accessibilityRole="header"
            style={
              Object {
                "color": "#1F2532",
                "fontFamily": "IBMPlexSans-SemiBold",
                "fontSize": 45,
                "lineHeight": 54.375,
              }
            }
          >
            Fragegruppe 1
          </Text>
          <View
            accessibilityHint="Closes current page or modal."
            accessibilityLabel="close"
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="QuestionnaireModal_close"
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                accessible={false}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#1F2532",
                          "fontSize": 24,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    󰅖
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <RCTScrollView
          onScroll={[Function]}
          scrollEventThrottle={16}
        >
          <View>
            <View
              style={
                Object {
                  "marginBottom": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans-SemiBold",
                    "fontSize": 33.75,
                    "lineHeight": 43.125,
                    "marginBottom": 5,
                    "marginTop": 5,
                  }
                }
              >
                Das ist eine Datumsabfrage
              </Text>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="overlay"
              >
                <View
                  style={
                    Object {
                      "paddingHorizontal": 10,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 1,
                        "borderColor": "#86939e",
                        "flexDirection": "row",
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "height": 40,
                          "justifyContent": "center",
                          "marginVertical": 4,
                          "paddingRight": 4,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "overflow": "hidden",
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "justifyContent": "center",
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              selectable={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#242424",
                                    "fontSize": 24,
                                  },
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                  Object {
                                    "fontFamily": "FontAwesome",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                              testID="iconIcon"
                            >
                              
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                    <TextInput
                      editable={false}
                      placeholderTextColor="#86939e"
                      pointerEvents="none"
                      style={
                        Object {
                          "color": "#242424",
                          "flex": 1,
                          "fontSize": 18,
                          "minHeight": 40,
                        }
                      }
                      testID="chosenDate"
                      underlineColorAndroid="transparent"
                      value={null}
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#ff190c",
                        "fontSize": 12,
                        "margin": 5,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F8F9FA",
          }
        }
        testID="BottomBar"
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#DADADA",
              "borderRadius": 2.5,
              "height": 5,
              "marginHorizontal": 5,
              "width": "95%",
            }
          }
          testID="progressBar"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#1F2532",
                  "borderRadius": 2.5,
                  "height": 5,
                },
                Object {
                  "width": "10.526315789473683%",
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "justifyContent": "space-evenly",
              "marginVertical": 10,
              "textAlign": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityHint="Jump to previous question."
              accessibilityLabel="back"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 44,
                  "opacity": 1,
                  "position": "relative",
                  "width": 44,
                }
              }
              testID="BottomBar_back_btn"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#1F2532",
                              "fontSize": 24,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        󰁍
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityHint="Jumps to the next questions, or closes the modal if this was the last question."
              accessibilityLabel="Next question, even though question is not yet answered."
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="BottomBar_confirm_btn"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderRadius": 28,
                      "height": 52,
                      "margin": 7,
                      "overflow": "hidden",
                      "width": 52,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#1F2532",
                          "borderRadius": 28,
                          "height": 52,
                          "justifyContent": "center",
                          "width": 52,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontSize": 24,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        󰄬
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityHint="Jump to the next question."
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 44,
                  "opacity": 1,
                  "position": "relative",
                  "width": 44,
                }
              }
              testID="BottomBar_fwd_btn"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#1F2532",
                              "fontSize": 24,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        󰁔
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
